python-checks/src/main/java/org/sonar/python/checks/MethodShouldBeStaticCheck.java

```diff
@@ -53,12 +54,33 @@ public void init() {
   public void visitNode(AstNode node) {
     if (CheckUtils.isMethodDefinition(node) && !alreadyStaticMethod(node) && hasValuableCode(node)){
       String self = getFirstArgument(node);
-      if (self != null && !isUsed(node, self)){
+      if (self != null && !isUsed(node, self) &&!isDocStringsAndNotImplementedError(node)){
         getContext().createLineViolation(this, MESSAGE, node.getFirstChild(PythonGrammar.FUNCNAME));
       }
     }
   }
 
+  private static boolean isDocStringsAndNotImplementedError(AstNode funcDef) {
+    List<AstNode> statements = funcDef.getFirstChild(PythonGrammar.SUITE).getChildren(PythonGrammar.STATEMENT);
+
+    if (!statements.isEmpty()) {
+      List<AstNode> firstStatements = new ArrayList<>(statements);
+      firstStatements.remove(statements.size() - 1);
+      for (AstNode statement : firstStatements) {
+        if (!isDocstring(statement)) {
+          return false;
+        }
+      }
+
+      AstNode testStatement = statements.get(statements.size() - 1).getFirstDescendant(PythonGrammar.TEST);
```

Comment:
@ivandalbosco Why don't you check that statement is `raise`?
E.g. this code doesn't raise issue, while it should
```
    def foo(self):
        return NotImplementedError
```